py/stabilizer/stream.py

```diff
@@ -87,11 +88,23 @@ class StabilizerStream(asyncio.DatagramProtocol):
     }
 
     @classmethod
-    async def open(cls, local_addr, maxsize=1):
-        """Open a UDP socket and start receiving frames"""
+    async def open(cls, addr, port, broker, maxsize=1):
+        """Open a UDP multicast socket and start receiving frames"""
         loop = asyncio.get_running_loop()
-        transport, protocol = await loop.create_datagram_endpoint(
-            lambda: cls(maxsize), local_addr=local_addr)
+        sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
+
+        # We need to specify which interface to receive broadcasts from, or Windows may choose the
+        # wrong one. Thus, use the broker address to figure out our local address for the interface
+        # of interest.
+        if ipaddress.ip_address(addr).is_multicast:
+            print('Subscribing to multicast')
+            group = socket.inet_aton(addr)
+            iface = socket.inet_aton('.'.join([str(x) for x in get_local_ip(broker)]))
+            sock.setsockopt(socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP, group + iface)
```

Comment:
* also `REUSEADDR`
* and move L108-L115 before the `bind()` (no need for `get_extra_info()`)
* move L93 (`loop`) closer to the `loop.create_datagram_endpoint()` in L107
* shouldn't we still `bind((addr, port))`?

Reply:
We can't use `bind((addr, port))` on windows with a multicast address because Windows thinks the `addr` is invalid and unroutable even if its a multicast/broadcast address. Thus, we set it to '' to always refer to local binding. We handle interface routing in the lines above.

Reply:
Ack. But for the non-multicast case I'd rather bind to the specific IP then. Otherwise it remains unused.